src/pybdy/nemo_bdy_gen_c.py

```diff
@@ -21,235 +23,514 @@ class Boundary:
     _SOUTH = [1, -1, 1, -1, None, -2, 1, -1]
     _EAST = [1, -1, 1, -1, 1, -1, 2, None]
     _WEST = [1, -1, 1, -1, 1, -1, None, -2]
+    _SUPPORTED_GRIDS = ["t", "u", "v", "f"]
 
-    def __init__(self, boundary_mask, settings, grid):
+    def __init__(self, boundary_mask: np.ndarray, settings: dict, grid: str) -> None:
         """
-        Generate the indices for NEMO Boundary and returns a Grid object with indices.
+        Generate the indices for NEMO Boundary and return a Grid object with indices.
 
-        Paramemters
-        -----------
-        boundary_mask -- boundary mask
-        settings -- dictionary of setting values
-        grid -- type of the grid 't', 'u', 'v'
-        Attributes:
-        bdy_i -- index
-        bdy_r -- r index
+        Parameters
+        ----------
+        boundary_mask
```

Comment:
`boundary_mask: np.ndarray`

apply similar format to all vars in Parameters list